odl/discr/diff_ops.py

```diff
@@ -87,7 +87,7 @@ def __init__(self, space, axis=0, method='forward', padding_method=None,
             accuracy in the interior.
         """
         if not isinstance(space, DiscreteLp):
-            raise TypeError('space {!r} is not a DiscreteLp instance.'
+            raise TypeError('`space` {!r} is not a DiscreteLp instance.'
```

Comment:
I'm tending towards not using them since they require to exactly give the name of the parameter, which is unfortunate for subclassing. For example, in geometries, when we give a partition `motion_part` for the motion parameters, and in a subclass, the parameter is named `angle_part`, then consider these two error messages (made up):
```
ValueError: motion parameter partition must be two-dimensional.
```
against this:
```
ValueError: `motion_part` must be two-dimensional.
```
I'd say the first one can be more easily reconciled with a different parameter name since natural language is used. The second message is in full conflict with a different parameter name.